lua/entities/gmod_wire_expression2/base/compiler.lua

```diff
@@ -651,42 +651,57 @@ end
 
 function Compiler:InstrNUM(args)
 	self.prfcounter = self.prfcounter + 0.5
-	RunString("E2Lib.Compiler.native = function() return " .. args[3] .. " end")
-	return { Compiler.native }, "n"
+	local num = tonumber(args[3])
+
+	return {function()
+		return num
+	end}, "n"
 end
 
 function Compiler:InstrNUMI(args)
 	self.prfcounter = self.prfcounter + 1
-	Compiler.native = { 0, tonumber(args[3]) }
-	RunString("local value = E2Lib.Compiler.native E2Lib.Compiler.native = function() return value end")
-	return { Compiler.native }, "c"
+	local value = {0, tonumber(args[3])}
```

Comment:
`NUMI`/`NUMJ`/`NUMK` are trivially okay, as they didn't do any parsing before and they still don't now.